eth/downloader/downloader.go

```diff
@@ -1068,18 +1067,17 @@ func (d *Downloader) findAncestor(p *peer) (uint64, error) {
 				}
 				start = check
 
+			case <-timeout:
+				glog.V(logger.Debug).Infof("%v: search header timeout", p)
+				return 0, errTimeout
+
 			case <-d.bodyCh:
```

Comment:
Ctrl+C, Ctrl+V: Same as above, ancestor lookup is ran before the fetchers start, so it's safe to deplete all data delivery channels.